adapters/httpapi_winhttp.c

```diff
@@ -128,26 +129,37 @@ HTTPAPI_RESULT HTTPAPI_Init(void)
 
     if (nUsersOfHTTPAPI == 0)
     {
-        if ((g_SessionHandle = WinHttpOpen(
-            NULL,
-#if defined _MSC_VER && _MSC_VER < 1600
-            WINHTTP_ACCESS_TYPE_DEFAULT_PROXY,
-#else
-            WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,
+        g_SessionHandle = NULL;
+#ifdef WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY
+        g_SessionHandle = WinHttpOpen(
+          NULL,
+          WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,
+          WINHTTP_NO_PROXY_NAME,
+          WINHTTP_NO_PROXY_BYPASS,
+          0);
 #endif
-            WINHTTP_NO_PROXY_NAME,
-            WINHTTP_NO_PROXY_BYPASS,
-            0)) == NULL)
+        if (g_SessionHandle == NULL)
         {
-            LogErrorWinHTTPWithGetLastErrorAsString("WinHttpOpen failed.");
-            result = HTTPAPI_INIT_FAILED;
+	    g_ProxyAutomatic = 0;
+            /*Automatic option is not supported on Windows 7 or earlier so try older option*/
+            g_SessionHandle = WinHttpOpen(
+                NULL,
+                WINHTTP_ACCESS_TYPE_DEFAULT_PROXY,
+                WINHTTP_NO_PROXY_NAME,
+                WINHTTP_NO_PROXY_BYPASS,
+                0);
         }
-        else
+        if (g_SessionHandle)
```

Comment:
Please revert this if block order change to minimize the code changes.

Reply:
I don't think I can do that without breaking my fix.  I have tried to do minimal changes.  If I can re-write this code, perhaps I can make it clearer.